### PR TITLE
fix: #144 initialize package iff <27.1

### LINKED ISSUE #123: Symbol's function definition is void: package-initialize
I have gotten the following warning since switching back to Emacs' development version.

```
Warning (package): Unnecessary call to ‘package-initialize’ in init file
```

It seems that our call of this is unnecessary.

>The reason automatic package loading occurs after loading the init file is that user options only receive their customized values after loading the init file, including user options which affect the packaging system. In some circumstances, you may want to load packages explicitly in your init file (usually because some other code in your init file depends on a package). In that case, your init file should call the function `package-initialize`. It is up to you to ensure that relevant user options, such as `package-load-list`, are set up prior to the `package-initialize` call. This will automatically set `package-enable-at-startup` to `nil`, to avoid loading the packages again after processing the init file. Alternatively, you may choose to completely inhibit package loading at startup, and invoke the command <kbd>M-x package-initialize</kbd> to load your packages manually.

Given that we require any dependencies that we need to configure before configuring them, do we still need this call? Emacs is saying no.

https://github.com/kotct/dot/blob/61b9eb5833b5f31bddfc176ffead20d29c256497/.emacs.d/init.el#L6-L7

### LINKED ISSUE #144: Make package-initialize conditional on package being uninitialized.
This is to avoid a [new warning because of 27.1](https://github.com/jkitchin/scimax/issues/194#issuecomment-385437906).

### PR DESCRIPTION
**Pull request by samontea**
This is to fix a [warning introduced in emacs 27.1](https://github.com/jkitchin/scimax/issues/194#issuecomment-385437906).

EDITs by @rye:
Closes #144. 
Closes #123.

### CODE REVIEW

**Review by rye — COMMENTED** [review]
Beautiful, thanks! We also have a `(package-initialize)` line in `init.el`, could you also do the same there?

**Comment by rye** [comment]
I have edited the OP to include closes statements for #144 and also #123 which is basically the same issue.

It may also be worth noting that [`package-initialize` is not really necessary anymore](https://git.savannah.gnu.org/cgit/emacs.git/tree/etc/NEWS?id=444930104e09b92ac27092aea44985637a39c26d#n172):

> Installed packages are now activated *before* loading the init file. As a result of this change, it is no longer necessary to call 'package-initialize' in your init file.

So it may be worth locking our initialization behind a version gate as well:

```elisp
(when (version< emacs-version "27.1")
  (package-initialize))
```

I'm… pretty sure that if we `package-initialize` in that way in our init file, the other `package-initialize` calls can be safely removed.

**Comment by samontea** [comment]
I'd prefer to use the variable `package--intialized` because even independent of these changes [multiple `package-initialize` calls are unadvisable](https://github.com/emacs-mirror/emacs/blob/a4a3c92e9de59bd0251f36326375cce898919edc/lisp/emacs-lisp/package.el#L1460-L1465).

**Comment by samontea** [comment]
haha jk. I switched to version checking. `package--intialized` didn't do what I wanted it to.

File: .emacs.d/init.el [hunk 1]
```diff
@@ -3,8 +3,10 @@
 ;; load any absolute must-have code
 (load "~/.emacs-preload" 'noerror 'nomessage 'nosuffix)
 
+(require 'package)
```
Comment (rye) [inline .emacs.d/init.el:6]: I'm not sure if we actually need to `(require 'package)`, but I have no issues with doing so besides potential performance losses.

**Review by rye — APPROVED** [review]

File: .emacs.d/init.el (hunk 1 above)
Comment (samontea) [reply .emacs.d/init.el:6]: it errored when i didn’t ¯\\_(ツ)_/¯ you can see that we’ve had to do this in [other places](https://github.com/kotct/dot/commit/accd55d87513bcbc4a158d27f55b7b456270ae5d#diff-9ff524a8cb0b974bb13c37be40048a4eL8)

**Pull request merged**